File: patterns/c-patterns/pattern127.c
// Pattern 127: ...

//  ***   ***
// ***** *****
// ***********
//  *********
//   *******
//    *****
//     ***
//      *

#include <stdio.h>

int main()
{
    int n = 6;
    int i, j;

    // Top part of the heart
    for (i = n / 2; i <= n; i += 2)
    {
        for (j = 1; j < n - i; j += 2)
        {
            printf(" ");
        }
        for (j = 1; j <= i; j++)
        {
            printf("*");
        }
        for (j = 1; j <= n - i; j++)
        {
            printf(" ");
        }
        for (j = 1; j <= i; j++)
        {
            printf("*");
        }
        printf("\n");
    }

    // Bottom part of the heart
    for (i = n; i >= 1; i--)
    {
        for (j = i; j < n; j++)
        {
            printf(" ");
        }
        for (j = 1; j <= (i * 2) - 1; j++)
        {
            printf("*");
        }
        printf("\n");
    }

    return 0;
}

File: patterns/javascript-patterns/pattern127.js
// Pattern 127: ...

//  ***   ***
// ***** *****
// ***********
//  *********
//   *******
//    *****
//     ***
//      *

const n = 6;

// Top part of the heart
for (let i = n / 2; i <= n; i += 2) {
    let row = '';

    for (let j = 1; j < n - i; j += 2) {
        row += ' ';
    }

    for (let j = 1; j <= i; j++) {
        row += '*';
    }

    for (let j = 1; j <= n - i; j++) {
        row += ' ';
    }

    for (let j = 1; j <= i; j++) {
        row += '*';
    }

    console.log(row);
}

// Bottom part of the heart
for (let i = n; i >= 1; i--) {
    let row = '';

    for (let j = i; j < n; j++) {
        row += ' ';
    }

    for (let j = 1; j <= (i * 2) - 1; j++) {
        row += '*';
    }

    console.log(row);
}

